### PR TITLE
chore: fix fhe=disable inference

### PR DESCRIPTION
**Pull request by jfrery**
closes https://github.com/zama-ai/concrete-ml-internal/issues/4076
closes https://github.com/zama-ai/concrete-ml-internal/issues/2604

### CODE REVIEW

File: src/concrete/ml/sklearn/base.py [hunk 1]
```diff
@@ -1382,7 +1382,17 @@ def compile(self, *args, **kwargs) -> Circuit:
     def _inference(self, q_X: numpy.ndarray) -> numpy.ndarray:
         assert self._tree_inference is not None, self._is_not_fitted_error_message()
 
-        return self._tree_inference(q_X)[0]
+        # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4077
```
Comment (kcelia) [inline src/concrete/ml/sklearn/base.py:1385]: I need some clarifications please:
You said: Ensembles of trees are cut right before the reduce sum node. But here we can see a + 0.5 right after which hasn't been added to the post_processing.
But you haven't changed the post_processing, why ?

Comment (kcelia) [reply src/concrete/ml/sklearn/base.py:1385]: you said: a batch of example in the disable still gives wrong predictions vs. one example at a time.

Here you are looping through each element at a time, isn't enough to fix the issue ?

Comment (jfrery) [reply src/concrete/ml/sklearn/base.py:1385]: it's the last fix for 4096 so it's closing it but the add node thing is tackled in #352 indeed.

Comment (kcelia) [reply src/concrete/ml/sklearn/base.py:1385]: Thanks !

**Comment by jfrery** [comment]
Ok so it seems the batch problem I observed came from torch 2.0. Closing this PR.

**Pull request closed without merging**